skimage/feature/scattering.py

```diff
@@ -0,0 +1,286 @@
+            Filters in the Fourier domain, stacked as (L,N,N) where
+            L is the num. of filters to apply
+        Returns
+        -------
+        filtered_signals : ndarray
+            Filtered signals in the Fourier domain,
+            stacked as (num_signals,L,N,N)
+        """
+    filtered_s = signals[:, np.newaxis, :, :]*filters[np.newaxis, :, :, :]
+    return filtered_s
+
+
+def _subsample(X, j):
+    """Spatial subsampling on the last two dimensions of X at a rate that
+    depends on 2**j.
+    Parameters
+    ----------
+    X : array like variable.
+        3D dnarray with shape (N,L,n,n) or (N,n,n). The subsampling is produced
+         in the last two dimensions (n,n)
+    j : int
+        Rate of subsampling is 2**j
+    Returns
+    -------
+    XX : ndarray
+        Subsampled images
+    """
+    dsf = 2**j
+    return dsf*X[..., ::dsf, ::dsf]
+
+
+def _apply_lowpass(img, phi, J, n_scat):
+    """Applies a low pass filter on the 'img' set of 2D arrays and subsamples.
+    Convolution the filter phi (in the Fourier domain) on the set of images
+    defined by img. The convolution is done in the Fourier domain, thus it is
+    a pointwise multiplication of the filter phi and the images, stored in the
+    last 2 dimensions of img. Then, the images are subsampled at the appropiate
+    rate to have 'n_scat' spatial coefficients
+    Parameters
+    ----------
+    img : ndarray of (N,L,n,n) or (N,n,n) shape.
+        Input images in the spatial domain, stacked along the first dimensions.
+    phi : Low pass filter as a 2D ndarray
+        Low pass filter in the Fourier domain, stored as (n,n) matrix
+    J : int
+        Rate of subsampling 2**J
+    n_scat : int
+        number of spatial coefficients of the scattering vector
+    Returns
+    -------
+    XX : ndarray
+        stacked images after being filtered and subsampled
+    """
+    img_filtered = np.real(np.fft.ifft2(np.fft.fft2(img)*phi))
+    n_nolp = img.shape[-1]
+    ds = int(n_nolp / n_scat)
+    return 2 ** (J - 1) * img_filtered[..., ::ds, ::ds]
+
+
+def scattering(x, wavelet_filters=None, m=2):
+    """Compute the scattering transform of a signal (or set of signals).
+    Given 'x', a set of 2D signals, this function computes the
+    scattering transform of these signals using the filter bank
+    'wavelet_filters'.
+    Notes
+    -----
+    **Bondary values: The scattering transform applies a set of convolutions
+    to the input signals. These convolutions are computed as the point-wise
+    multiplication in the Fourier domain, thus the boundary values of the
+    image are circular or cyclic. In case you need other
+    kind of boundary values, for instance zero-padded, you should
+    edit the images before calling this function.
+    **Shape of x: The signals x must be squared shaped, thus (N,px,px)
+    and not (N,px,py) for py != px. In case the images are rectangular,
+    they will be cropped to the smallest dimension, px or py.
```

Comment:
Well, same reason as before, we dont want to start resizing the image, because that changes its geometry, so if the user wants to do it, that is fine, they can do it consciously. Again, resizing the image is easy and can be done with skimage.